src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java

```diff
@@ -198,8 +198,10 @@ public Flow getFlow(String flowId) throws FlowNotFoundException {
             flows.addAll(flowRepository.findByEndpointSwitch(switchId));
         }
         // need to return Flows unique by id
+        // Due to possible race condition we can have one flow with different flow paths
+        // In this case we should get the last one
```

Comment:
HashSet ... is unordered, how do you propose to define "the last one"?

Reply:
Reworked code to remove intermediate set at all.